src/chain.ts

```diff
@@ -0,0 +1,169 @@
+import { Ucan, Capability, Fact } from "./types"
+import * as token from "./token"
+
+
+/**
+ * Represents a deeply verified chain of UCANs.
+ *
+ * These chains can actually be trees sometimes (if a ucan has >1 proofs).
+ */
+export class Chained {
```

Comment:
I like the class-based semantics here & think it would fit well in token also.

I often find myself wanting to do
```ts
const token = ucan.build(....)
const encoded = token.encode()
// instead of
const encoded = ucan.encode(token)
```

Reply:
Yeah! They're somewhat syntactically nicer, I agree.
I usually don't like classes in javascript, because to test whether something is a class you'd use `instanceof`. But often bundlers get in your way and you can often accidentally end up with some code in multiple places. E.g. you've got `(await import("ucans")).Chained != (await import("ucans/chained")).Chained` and as such your `instanceof` can break with one of them.

In this case I still went with it. I'm *still* not sure whether that's a good idea, but the main reason was that I wanted something with conceptually private data. `Chained` is supposed to be opaque, that's not the case with the token stuff in `token.ts`.

I've done some further work on the indexer yesterday evening and realized that I *do* want to have a builder API. That one will make use of class-based dot syntax such that building UCANs will be much more fun. :)